linera-service/src/client.rs

```diff
@@ -0,0 +1,1002 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::config::WalletState;
+use anyhow::{Context, Result};
+use async_graphql::InputType;
+use linera_base::{
+    abi::ContractAbi,
+    identifiers::{ChainId, MessageId, Owner},
+};
+use linera_execution::Bytecode;
+use once_cell::sync::OnceCell;
+use serde::ser::Serialize;
+use serde_json::{json, value::Value};
+use std::{
+    collections::{BTreeMap, HashMap, HashSet},
+    env, fs,
+    ops::RangeInclusive,
+    path::{Path, PathBuf},
+    process::Stdio,
+    rc::Rc,
+    str::FromStr,
+    time::Duration,
+};
+use tempfile::{tempdir, TempDir};
+use tokio::{
+    process::{Child, Command},
+    sync::Mutex,
+};
+use tonic_health::proto::{
+    health_check_response::ServingStatus, health_client::HealthClient, HealthCheckRequest,
+};
+use tracing::{info, warn};
+
+/// The name of the environment variable that allows specifying additional arguments to be passed
+/// to `cargo` when starting client, server and proxy processes.
+pub const CARGO_ENV: &str = "INTEGRATION_TEST_CARGO_PARAMS";
+
+/// The name of the environment variable that allows specifying additional arguments to be passed
+/// to the binary when starting a server.
+const SERVER_ENV: &str = "INTEGRATION_TEST_SERVER_PARAMS";
+
+/// The name of the environment variable that allows specifying additional arguments to be passed
+/// to the node-service command of the client.
+const CLIENT_SERVICE_ENV: &str = "INTEGRATION_TEST_CLIENT_SERVICE_PARAMS";
```

Comment:
Maybe these shouldn't be called `INTEGRATION_TEST_*` anymore, since they now also apply to `linera net up`. Maybe `LOCAL_NET_*`?